content/blog/scale-kubernetes-clusters-using-hpe-greenlake-terraform-provider.md

```diff
@@ -0,0 +1,394 @@
+date: 2023-05-12T04:07:48.888Z
+author: Tanushi Agarwal
+authorimage: /img/photoforblog.jpg
+disable: false
+---
+IaC or Infrastructure as code is an practice of automating the process of managing and provisioning infrastructure through code instead of manual processes. It gives organizations tools to create, manage, and destroy compute resources by statically defining and declaring these resources in codeblocks. It helps to increase operational agility, simplify management, reduce errors, and save cost.
+
+> I﻿n this post, I will explore options to declare and upgrade Kubernetes clusters on HPE GreenLake using the HPE GreenLake Terraform Provider.
+
+A﻿s the part of upgrade, the following 3 scenarios are supported --  
+
+1. Scaling of a cluster's worker nodes. Please refer blog post [Scale Kubernetes Clusters using HPE GreenLake Terraform Provider](https://developer.hpe.com/blog/scale-kubernetes-cluster-using-hpe-greenlake-terraform-provider/) to check out scaling options available for woker nodes. 
+2. Upgrade the kubernetes version of the cluster. Covered in this blog.
+3. Upgrade the OS version of the worker nodes inside a cluster. Covered in this blog.
+
+# Prerequisite
+
+Before starting this tutorial, it is recommended that you read the blog post  [Kubernetes Cluster as Code - Part 1](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), which includes steps for creating a Kubernetes cluster. This post expands upon that scenario by examining how to upgrade a cluster's kubernetes version & OS version of its woker nodes. 
+
+# Verify existing Kubernetes cluster
+
+After the cluster is created following the instructions found in the  [Kubernetes Cluster as Code - Part 1 blog post](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), launch HPE GreenLake Central console and verify that the cluster is present under the appropriate tenant.
+
+You should see the  **tf-test**  cluster present under  **Dashboard -> Manage your Private Cloud -> Containers**.
+
+![](https://developer.hpe.com/img/cluster_list.jpg)
+
+Below is the reference Terraform configuration file for the existing cluster.
+
+```hcl
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+ 
+variable "HPEGL_SPACE" {
+  type = string
+}
+ 
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = var.HPEGL_SPACE
+ }
+ 
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+ 
+resource hpegl_caas_cluster test {
+  name         = "tf-test"
+  blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+  site_id      = data.hpegl_caas_site.blr.id
+  space_id     = var.HPEGL_SPACE
+}
+```
+
+# Kubernetes version upgrade
+
+For Kubernetes version upgrade, we need to specify the new version of kubernetes that is available for upgrade in the resources block. 
```

Comment:
```suggestion
For the Kubernetes version upgrade, we need to specify the new version of Kubernetes that is available for upgrade in the resources block. 
```

Reply:
Suggest to use "you need to specify,..." instead of "we need to specify...".